sqsjobs/item.go

```diff
@@ -148,9 +156,9 @@ func (i *Item) Ack() error {
 	return nil
 }
 
-func (i *Item) Nack() error {
+func (i *Item) commonNack(requeue bool, delay int) error {
 	if atomic.LoadUint64(i.Options.stopped) == 1 {
-		return errors.Str("failed to acknowledge the JOB, the pipeline is probably stopped")
+		return errors.Str(pipelineStoppedError)
 	}
 	defer func() {
 		i.Options.cond.Signal()
```

Comment:
yeah, it should be called in all cases. This tells our internal counter to decrement the number of active messages in flight (aka consumed from listener and pushed to the priority queue). Since we're done with that message anyway (delete after pushing it back to SQS), it is OK to decrement that number.

Reply:
Ok, so it was a bug that it was not called before, on these lines:

```go
func (i *Item) NackWithOptions(requeue bool, delay int) error {
	if atomic.LoadUint64(i.Options.stopped) == 1 {
		return errors.Str("failed to acknowledge the JOB, the pipeline is probably stopped")
	}

	// message already deleted
	if i.Options.AutoAck {
		i.Options.cond.Signal()
		atomic.AddInt64(i.Options.msgInFlight, ^int64(0))
		return nil
	}

	if requeue {
                // it was not called in here!
		// requeue message
		err := i.Requeue(nil, delay)
		if err != nil {
			return err
		}
              
		return nil
	}
```

Reply:
Yeah, I guess that was a bug. I incremented the counter in the listener.go, after requeue, we still had +1 message + another one when the same message read in the listener.go.